src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/Strings.vb

```diff
@@ -3,48 +3,280 @@
+
+            End Try
+
         End Function
-        Public Function AscW([string] As String) As Integer
-            If ([string] Is Nothing) OrElse ([string].Length = 0) Then
-                Throw New Global.System.ArgumentException()
+
+        Public Function Asc(ByVal [String] As String) As Integer
+
+            If ([String] Is Nothing) OrElse ([String].Length = 0) Then
+                Throw New ArgumentException(SR.Argument_LengthGTZero1, NameOf([String]))
             End If
-            Return AscW([string].Chars(0))
+
+            Dim ch As Char = [String].Chars(0)
+            Return Asc(ch)
+
         End Function
-        Public Function AscW([string] As Char) As Integer
-            Return AscW([string])
+
+        Public Function AscW([String] As String) As Integer
+            If ([String] Is Nothing) OrElse ([String].Length = 0) Then
+                Throw New Global.System.ArgumentException(SR.Argument_LengthGTZero1, NameOf([String]))
+            End If
+            Return AscW([String].Chars(0))
+        End Function
+
+        Public Function AscW([String] As Char) As Integer
+            Return AscW([String])
+        End Function
+
+        Public Function Chr(ByVal CharCode As Integer) As Char
+            ' Documentation claims that < 0 or > 255 gives an ArgumentException
+            If CharCode < -32768 OrElse CharCode > 65535 Then
+                Throw New ArgumentException(SR.Argument_RangeTwoBytes1, NameOf(CharCode))
+            End If
+
+            If CharCode >= 0 AndAlso CharCode <= 127 Then
+                Return Convert.ToChar(CharCode)
+            End If
+
+            Try
+                Dim enc As Encoding
+
+                enc = Encoding.GetEncoding(GetLocaleCodePage())
+
+                If enc.IsSingleByte Then
+                    If CharCode < 0 OrElse CharCode > 255 Then
+                        Throw VbMakeException(vbErrors.IllegalFuncCall)
+                    End If
+                End If
+
+                Dim dec As Decoder
+                Dim CharCount As Integer
+                Dim c(1) As Char 'Use 2 char array, but only return first Char if two returned
+                Dim b(1) As Byte
+
+                dec = enc.GetDecoder()
+                If CharCode >= 0 AndAlso CharCode <= 255 Then
+                    b(0) = CByte(CharCode And &HFFS)
+                    CharCount = dec.GetChars(b, 0, 1, c, 0)
+
+                Else
+                    'Bytes must be swapped in memory to HI/LO
+                    b(0) = CByte((CharCode And &HFF00I) >> 8)
+                    b(1) = CByte(CharCode And &HFFI)
+                    CharCount = dec.GetChars(b, 0, 2, c, 0)
+
+                End If
+
+                'VB6 ignored the lobyte if it hibyte was not a valid lead character
+                'CharCount will be zero if the hibyte was not a lead character
+
+                Return c(0)
+
+            Catch ex As Exception
+                Throw ex
+            End Try
```

Comment:
Same try/catch question.

Reply:
I have no idea. Figured it was best to leave as is rather than change some behavior subtlely. Seems a non goal to attempt to fix this code as it is purely to allow existing libraries to run.

Reply:
Ok